### ISSUE TITLE
question about the structure and loss of the pytorch implementation

### ISSUE THREAD

**Issue by XinjieInformatik** [biubug6/Pytorch_Retinaface#21]
hi @biubug6 , thanks for sharing your code! The pytorch version is much more clear for me compared the MXnet version. May I ask you, do you keep the same network structure and loss function in this pytorch version as the original MXnet?

**Comment by biubug6** [comment]
The implementation of pytorch is consistent with the original MXnet in network structure and loss function. It has slight difference in data preprocessing and sample selection. Experiments show that these changes slightly get better result. You can see the code for details.

**Comment by XinjieInformatik** [comment]
Understand, thanks for your rely!

**Comment by XinjieInformatik** [comment]
`    def _make_class_head(self,fpn_num=3,inchannels=64,anchor_num=2):
        classhead = nn.ModuleList()
        for i in range(fpn_num):
            classhead.append(ClassHead(inchannels,anchor_num))
        return classhead
    
    def _make_bbox_head(self,fpn_num=3,inchannels=64,anchor_num=2):
        bboxhead = nn.ModuleList()
        for i in range(fpn_num):
            bboxhead.append(BboxHead(inchannels,anchor_num))
        return bboxhead

    def _make_landmark_head(self,fpn_num=3,inchannels=64,anchor_num=2):
        landmarkhead = nn.ModuleList()
        for i in range(fpn_num):
            landmarkhead.append(LandmarkHead(inchannels,anchor_num))
        return landmarkhead `

Hi @biubug6 , seens in the code, the head layers haven't shared across the pyramid levels?

but in retinaface paper
> We employ a shared loss head (1×1conv) across different feature mapsHn×Wn×256,n∈{2,...,6}.
also for retinanet, the parameters of the subnet are shared across all pyramid levels.

**Comment by biubug6** [comment]
You are very careful. But the MXnet version hasn't been shared loss head and I just keep in line with the official implementation. In fact,  no shared loss head would get better results.
